### PR TITLE
feat(server/list): group servers by type

### PR DESCRIPTION
**Pull request by djohts**

### CODE REVIEW

**Review by WilliamDavidHarrison — DISMISSED** [review]
LGTM, but I will wait for @DEV-DIBSTER's approval first.

**Comment by djohts** [comment]
Will cleanup some repetitive code tomorrow and then reopen it :)

**Review by DEV-DIBSTER — DISMISSED** [review]

**Review by WilliamDavidHarrison — APPROVED** [review]

**Pull request merged**
